### ISSUE TITLE
TypeError: No overloads for "dict(a=True, b=False)"

### ISSUE THREAD

**Issue by giyyapan** [microsoft/pyright#756]
**Describe the bug**
Statement `d = dict(a=True, b=False)` is raising error.
![image](https://user-images.githubusercontent.com/3479453/85520492-948cd180-b635-11ea-9ffc-40f5b14eb8fa.png)

**To Reproduce**
```python
some_dict = dict(a=True, b=False)
```

**Expected behavior**
No error reported

**VS Code extension or command-line**
VSCode extension version 1.1.45.

**Additional context**
This issue is originally found in our project on statements below:
```python
        s_notify_data = dict(
            is_staff_notify=True,
            staff_notify_status="stop",
            staff_notify_days=7,
            staff_notify_hour=9,
            staff_notify_minute=0,
            is_admin_notify=False,
            admin_notify_status="stop",
            admin_notify_days=14,
            admin_notify_hour=9,
            admin_notify_minute=0
        )
```
And I was able to simplify it to `dict(a=True, b=False)`, but not sure if the original statements reveal other bugs.

Thank you.

**Comment by erictraut** [comment]
Thanks for reporting the error. I'm embarrassed to say that this was a recent regression that wasn't caught by my test cases. It will be fixed in the next release of Pyright.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.46, which I just published.
